recipes/pybind11/all/conanfile.py

```diff
@@ -50,24 +43,45 @@ def package(self):
         self.copy("LICENSE", src=self._source_subfolder, dst="licenses")
         cmake = self._configure_cmake()
         cmake.install()
-        lib_folder = os.path.join(
-            self.package_folder, "lib", "cmake", "pybind11")
-        os.rename(
-            os.path.join(lib_folder, "pybind11Config.cmake"),
-            os.path.join(lib_folder, "pybind11Install.cmake"))
-        os.unlink(os.path.join(lib_folder, "pybind11ConfigVersion.cmake"))
+        for filename in ["pybind11Targets.cmake", "pybind11Config.cmake", "pybind11ConfigVersion.cmake"]:
+            try:
+                os.unlink(os.path.join(self.package_folder, "lib", "cmake", "pybind11", filename))
+            except:
+                pass
+        if tools.Version(self.version) >= "2.6.0":
+            tools.replace_in_file(os.path.join(self.package_folder, "lib", "cmake", "pybind11", "pybind11Common.cmake"),
+                                  "if(TARGET pybind11::lto)",
+                                  "if(FALSE)")
+            tools.replace_in_file(os.path.join(self.package_folder, "lib", "cmake", "pybind11", "pybind11Common.cmake"),
+                                  "add_library(",
+                                  "# add_library(")
 
     def package_id(self):
         self.info.header_only()
 
     def package_info(self):
-        self.cpp_info.includedirs.append(os.path.join(
-            self.package_folder, "include", "pybind11"))
-
         cmake_base_path = os.path.join("lib", "cmake", "pybind11")
-        self.cpp_info.builddirs = [cmake_base_path]
+        if tools.Version(self.version) >= "2.6.0":
+            self.cpp_info.components["main"].names["cmake_find_package"] = "pybind11"
```

Comment:
I comment I see often (but I am no expert) we need to be caution about generic conan component names

Reply:
yes, already mentioned here: https://github.com/conan-io/conan-center-index/pull/4445#discussion_r577760664